documentation/homework/HomeWork3/todo-app/src/App.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import AddTaskForm from './components/AddTaskForm';
+import TaskList from './components/TaskList';
+import Separator from './components/Separator';
+import DeletedTaskList from './components/DeletedTaskList';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      tasks: [],
+      deleted: []
+    }
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.toggleComplete = this.toggleComplete.bind(this);
+    this.deleteTask = this.deleteTask.bind(this);
+    this.undoDelete = this.undoDelete.bind(this);
+    this.deleteForever = this.deleteForever.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      value: e.target.value
+    });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const task = {
+      id: Date.now(),
+      name: this.state.value,
+      completed: false
+    };
+    this.setState(prevState =>
+      ({
+        tasks: prevState.tasks.concat(task),
+        value: ''
+      })
+    );
+  }
+
+  toggleComplete(task) {
+    task.completed = !task.completed;
```

Comment:
This is works fine I think, but you can take checkbox value when onChange right?
so something like input.target.value!